tests/integration/cloud/end_to_end/test_pandas_datasource.py

```diff
@@ -67,90 +63,60 @@ def datasource(
-            data_asset.build_batch_request()
-        batch_request = data_asset.build_batch_request(options={"dataframe": pandas_test_df})
-    else:
-        batch_request = data_asset.build_batch_request()
-    return batch_request
+def batch_definition(
+    context: CloudDataContext,
+    data_asset: DataFrameAsset,
+) -> BatchDefinition:
+    batch_def_name = f"batch_def_{uuid.uuid4().hex}"
+    return data_asset.add_batch_definition_whole_dataframe(
+        name=batch_def_name,
+    )
 
 
 @pytest.fixture(scope="module")
-def expectation_suite(
+def validation_definition(
     context: CloudDataContext,
     expectation_suite: ExpectationSuite,
-) -> ExpectationSuite:
-    """Add Expectations for the Data Assets defined in this module.
-    Note: There is no need to test Expectation Suite CRUD.
-    Those assertions can be found in the expectation_suite fixture.
-    """
-    expectation_suite.add_expectation_configuration(
-        expectation_configuration=ExpectationConfiguration(
-            type="expect_column_values_to_not_be_null",
-            kwargs={
-                "column": "string",
-                "mostly": 1,
-            },
+    batch_definition: BatchDefinition,
+) -> Generator[ValidationDefinition, None, None]:
+    validation_def_name = f"val_def_{uuid.uuid4().hex}"
+    yield context.validation_definitions.add(
+        ValidationDefinition(
+            name=validation_def_name,
+            data=batch_definition,
+            suite=expectation_suite,
         )
     )
-    return expectation_suite
+    context.validation_definitions.delete(name=validation_def_name)
 
 
-@pytest.mark.cloud
-def test_interactive_validator(
-    context: CloudDataContext,
-    validator: Validator,
-):
-    """Test interactive evaluation of the Data Assets in this module using an existing Validator.
-    Note: There is no need to test getting a Validator or using Validator.head(). That is already
-    tested in the validator fixture.
-    """
-    expectation_validation_result: ExpectationValidationResult = (
-        validator.expect_column_values_to_not_be_null(
-            column="datetime",
-            mostly=1,
-        )
-    )
-    assert expectation_validation_result.success
+@pytest.fixture(scope="module")
+def checkpoint(
+    checkpoint: Checkpoint,
+    validation_definition: ValidationDefinition,
+) -> Checkpoint:
+    checkpoint.validation_definitions = [validation_definition]
+    checkpoint.save()
+    return checkpoint
 
 
-@pytest.mark.xfail(reason="Fails due to a V1 change in the Checkpoint shape.", strict=True)
 @pytest.mark.cloud
-def test_checkpoint_run(checkpoint: Checkpoint):
+def test_checkpoint_run(checkpoint: Checkpoint, pandas_test_df):
     """Test running a Checkpoint that was created using the entities defined in this module."""
-    checkpoint_result: CheckpointResult = checkpoint.run()
+    checkpoint_result = checkpoint.run(batch_parameters={"dataframe": pandas_test_df})
```

Comment:
I'm confused why this works. When I dig deep into fixtures, it looks like we have this expectation: `ExpectColumnValuesToNotBeNull(column="name", mostly=1)` for a a dataframe with no `name` column.

Reply:
Update: I see it's failing CI for this reason 😄

Reply:
indeed